tests/at_functional_test/test/enroll_verb_test.dart

```diff
@@ -77,47 +79,364 @@ void main() {
+      print(updateResponse);
+      assert((!updateResponse.contains('Invalid syntax')) &&
+          (!updateResponse.contains('null')));
+    });
+
+    // Purpose of the tests
+    //  1. Authenticate and send the enroll request for wavi namespace
+    //  2. pkam using the enroll id
+    //  3. Already there is a atmosphere key in server
+    //  4. Do a llookup for the atmosphere key
+    //  5. Assert that the llookup throws an exception
+    test(
+        'enroll request on authenticated connection for wavi namespace and llookup for a atmosphere key',
+        () async {
+      await socket_writer(socketConnection1!, 'from:$firstAtsign');
+      var fromResponse = await read();
+      print('from verb response : $fromResponse');
+      fromResponse = fromResponse.replaceAll('data:', '');
+      var pkamDigest = generatePKAMDigest(firstAtsign, fromResponse);
+      await socket_writer(socketConnection1!, 'pkam:$pkamDigest');
+      var pkamResult = await read();
+      expect(pkamResult, 'data:success\n');
+
+      // Before creating a enroll request with wavi namespace
+      // create a atmosphere key
+      await socket_writer(
+          socketConnection1!, 'update:$atmosphereKey atmospherevalue');
+      var updateResponse = await read();
+      assert((!updateResponse.contains('Invalid syntax')) &&
+          (!updateResponse.contains('null')));
+
+      var enrollRequest =
+          'enroll:request:appName:wavi:deviceName:pixel:namespaces:[wavi,rw]:apkamPublicKey:${pkamPublicKeyMap[firstAtsign]!}\n';
+      await socket_writer(socketConnection1!, enrollRequest);
+      var enrollResponse = await read();
+      print(enrollResponse);
+      enrollResponse = enrollResponse.replaceFirst('data:', '');
+      var enrollJsonMap = jsonDecode(enrollResponse);
+      expect(enrollJsonMap['enrollmentId'], isNotEmpty);
+      expect(enrollJsonMap['status'], 'success');
+
+      var enrollmentId = enrollJsonMap['enrollmentId'];
+
+      // now do the apkam using the enrollment id
+      pkamDigest = generatePKAMDigest(firstAtsign, fromResponse);
+      var apkamEnrollId = 'pkam:enrollApprovalId:$enrollmentId:$pkamDigest\n';
+
+      await socket_writer(socketConnection1!, apkamEnrollId);
+      var apkamEnrollIdResponse = await read();
+      expect(apkamEnrollIdResponse, 'data:success\n');
+
+      await socket_writer(socketConnection1!, 'llookup:$atmosphereKey');
+      var llookupResponse = await read();
+      print(llookupResponse);
+      expect(llookupResponse,
+          'error:AT0009-UnAuthorized client in request : Enrollment Id: $enrollmentId is not authorized for local lookup operation\n');
+    });
+
+    // Purpose of the tests
+    //  1. Authenticate and send the enroll request for wavi namespace
+    //  2. pkam using the enroll id
+    //  3. Already there is a atmosphere key in server
+    //  4. Do a scan
+    //  5. Assert that the scan verb doesn't return the atmosphere key
+    test(
```

Comment:
add a positive test to lookup public keys

Reply:
Updated